Lib/test/test_syntax.py

```diff
@@ -1513,6 +1513,27 @@ def test_empty_line_after_linecont(self):
         except SyntaxError:
             self.fail("Empty line after a line continuation character is valid.")
 
+        # See issue-46091
+        s = r"""\
+def fib(n):
+    \
+'''Print a Fibonacci series up to n.'''
+    \
+a, b = 0, 1
+    \
+while a < n:
+        \
+print(a, end=' ')
+        \
+a, b = b, a+b
+    \
+print()
```

Comment:
This example is both too long and too short.

It doesn't really test anything more than a shorter example, say just the first three lines.

Moreover, it doesn't test that the code is parsed the same as the equivalent version without backslashes.

And it doesn't test the stated property quoted in bpo-46091 from the docs: "Indentation cannot be split over multiple physical lines using backslashes; the whitespace up to the first backslash determines the indentation."

Reply:
> And it doesn't test the stated property quoted in [bpo-46091](https://bugs.python.org/issue46091) from the [docs](https://docs.python.org/3/reference/lexical_analysis.html#indentation): "Indentation cannot be split over multiple physical lines using backslashes; the whitespace up to the first backslash determines the indentation."

Hummmm is testing the second part at least as it's checking that the whitespace up the the `\` determines the identation of the next line as opposed to something like:

```
pass
      \
      
pass
```

where the `\` and the `\n` will just produce an empty line

Reply:
Since we don't have access to the tokenizer, checking that the ASTs are the same is reasonable. In fact, you can probably just use ast.unparse() and compare the output with the equivalent without backslashes.

But this reminds me. Does tokenize.py have the same behavior?

Reply:
This is the tokenization of:

```

```

with `\`:

```
0,0-0,0:            ENCODING       'utf-8'
1,0-1,3:            NAME           'def'
1,4-1,7:            NAME           'fib'
1,7-1,8:            OP             '('
1,8-1,9:            NAME           'n'
1,9-1,10:           OP             ')'
1,10-1,11:          OP             ':'
1,11-1,12:          NEWLINE        '\n'
2,0-2,4:            INDENT         '    '
3,0-3,39:           STRING         '"""Print a Fibonacci series up to n."""'
3,39-3,40:          NEWLINE        '\n'
5,0-5,1:            NAME           'a'
5,1-5,2:            OP             ','
5,3-5,4:            NAME           'b'
5,5-5,6:            OP             '='
5,7-5,8:            NUMBER         '0'
5,8-5,9:            OP             ','
5,10-5,11:          NUMBER         '1'
5,11-5,12:          NEWLINE        '\n'
6,0-6,0:            DEDENT         ''
6,0-6,0:            ENDMARKER      ''
```

and without:

```
0,0-0,0:            ENCODING       'utf-8'
1,0-1,3:            NAME           'def'
1,4-1,7:            NAME           'fib'
1,7-1,8:            OP             '('
1,8-1,9:            NAME           'n'
1,9-1,10:           OP             ')'
1,10-1,11:          OP             ':'
1,11-1,12:          NEWLINE        '\n'
2,0-2,4:            INDENT         '    '
2,4-2,43:           STRING         '"""Print a Fibonacci series up to n."""'
2,43-2,44:          NEWLINE        '\n'
3,4-3,5:            NAME           'a'
3,5-3,6:            OP             ','
3,7-3,8:            NAME           'b'
3,9-3,10:           OP             '='
3,11-3,12:          NUMBER         '0'
3,12-3,13:          OP             ','
3,14-3,15:          NUMBER         '1'
3,15-3,16:          NEWLINE        '\n'
4,0-4,0:            DEDENT         ''
4,0-4,0:            ENDMARKER      ''
```

Reply:
But unfortunately the tokenization for:

```
pass
        \
pass
```

is wrong:

```
0,0-0,0:            ENCODING       'utf-8'
1,0-1,4:            NAME           'pass'
1,4-1,5:            NEWLINE        '\n'
2,0-2,4:            INDENT         '    '
3,0-3,1:            NEWLINE        '\n'
4,0-4,0:            DEDENT         ''
4,0-4,4:            NAME           'pass'
4,4-4,5:            NEWLINE        '\n'
5,0-5,0:            ENDMARKER      ''
```

but that should be fixed separately to this PR